### DISCUSSION TITLE
Router.push always does full page reload

### DISCUSSION THREAD

**Discussion by tomitrescak** [vercel/next.js#14300] [Help] [answered]
Hello, I have a url like `/item/[id]`
I have a button, which allows to add a new item, which triggers a modal asking for a name.
When I insert the new item into db via sync call, I redirect to the new URL `item/[newId]`.
I use `Router.push('/item/newId')`.

Unfortunately, this always attracts a full-page reload.
Is there any way to dot this "SPA-way" so that I would only redraw interface with new data?
It seriously breaks the experience.

thanks

**Comment by desaijay315** [comment]
You can check the example here - https://github.com/desaijay315/movie_list_nextjs_app

**Reply by tomitrescak** [reply]
thanks for the effort,  but I found the answer without having to go through whole repository.

**Comment by tomitrescak** [comment, marked as answer]
And just like that I found the answer. I was using `.push` the wrong way:

I did:

`Router.push('/item/123')` instead of `Router.push('/item/[id]', '/item/123')`

**Reply by tomitrescak** [reply]
This is my source: 

https://nextjs.org/docs/api-reference/next/router

**Reply by desaijay315** [reply]
Awesome
